Rakefile

```diff
@@ -44,21 +44,82 @@ namespace :generate do
     path = Pathname("test/stdlib/#{klass}_test.rb")
     raise "#{path} already exists!" if path.exist?
 
-    path.write <<~RUBY
-      require_relative "test_helper"
-      
-      class #{klass}Test < StdlibTest
-        target #{klass}
-        # library "pathname", "set", "securerandom"     # Declare library signatures to load
-        using hook.refinement
-
-        # def test_method_name
-        #   # Call the method
-        #   method_name(arg)
-        #   method_name(arg, arg2)
-        # end
+    require "erb"
+    require "ruby/signature"
+
+    class TestTemplateBuilder
+      attr_reader :klass, :env
+
+      def initialize(klass)
+        @klass = klass
+
+        @env = Ruby::Signature::Environment.new
+        Ruby::Signature::EnvironmentLoader.new.load(env: @env)
+      end
+
+      def call
+        ERB.new(<<~ERB, trim_mode: "-").result(binding)
+          require_relative "test_helper"
+
+          class <%= klass %>Test < StdlibTest
+            target <%= klass %>
+            # library "pathname", "set", "securerandom"     # Declare library signatures to load
+            using hook.refinement
+          <%- class_methods.each do |method_name, definition| %>
+            def test_class_method_<%= test_name_for(method_name) %>
+            <%- definition.method_types.each do |method_type| -%>
+              # <%= method_type %>
+              <%= klass %>.<%= method_name %>
+            <%- end -%>
+            end
+          <%- end -%>
+          <%- instance_methods.each do |method_name, definition| %>
+            def test_<%= test_name_for(method_name) %>
+            <%- definition.method_types.each do |method_type| -%>
+              # <%= method_type %>
+              <%= klass %>.new.<%= method_name %>
+            <%- end -%>
+            end
+          <%- end -%>
+          end
+        ERB
+      end
+
+      private
+
+      def test_name_for(method_name)
```

Comment:
This also needs `<=>`, `<=`, `>=`, `>`, `<`, `+`, `-`, `*`, `/`, `**`, etc. There's a table on this page with a bunch of the special method names: https://ruby-doc.org/core-2.4.0/doc/syntax/methods_rdoc.html

Reply:
Something like this?:
```ruby
special_methods_map = {
  '==': 'double_equal',
  '!=': 'not_equal'
  '===': 'triple_equal',
  '[]': 'square_bracket',
  '[]=': 'square_bracket_assign',
  '>': 'greater_than',
  '<': 'less_than',
  '>=': 'greater_than_equal_to',
  '<=': 'less_than_equal_to',
  '<=>': 'spaceship',
  '+': 'plus',
  '-': 'minus',
  '*': 'multiply',
  '/': 'divide',
  '**': 'power',
  '%': 'modulus',
  '&': 'and',
  '^': 'xor',
  '>>': 'right_shift',
  '<<': 'left_shift',
  '=~': 'pattern_match'
  '!~': 'does_not_match'
}

return special_methods_map[method_name] if special_methods_map.keys.include?(method_name)

return method_name
```

Reply:
I'm wondering if defining test methods via `define_method` is acceptable

```ruby
define_method "test==" do
end
```

Reply:
Using `define_method` would work, but I think adding renamings here for them too makes more sense. (Choosing `def` + renaming or `define_method` would look like strange to me. We already have _renaming table_ here. etc...)

Reply:
I updated name mapping